lib/decorate.js

```diff
@@ -9,6 +9,14 @@ function decorate (name, fn, dependencies) {
     checkDependencies(this, dependencies)
   }
 
+  if (fn.hasOwnProperty('getter') || fn.hasOwnProperty('setter')) {
```

Comment:
I would use `typeof fn.getter === 'function'` (same for the *setter*), in this way we can be sure that all the other possible cases can be used normally (strings, numbers, objects...).